adapters/pubnative/pubnative.go

```diff
@@ -0,0 +1,172 @@
+package pubnative
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type PubnativeAdapter struct {
+	URI string
+}
+
+func (a *PubnativeAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	impCount := len(request.Imp)
+	requestData := make([]*adapters.RequestData, 0, impCount)
+	errs := []error{}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	err := checkRequest(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	for _, imp := range request.Imp {
+		requestCopy := *request
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		var pubnativeExt openrtb_ext.ExtImpPubnative
+		if err := json.Unmarshal(bidderExt.Bidder, &pubnativeExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		err := convertImpression(&imp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		requestCopy.Imp = []openrtb.Imp{imp}
+		reqJSON, err := json.Marshal(&requestCopy)
+		if err != nil {
+			errs = append(errs, err)
+			return nil, errs
+		}
+
+		reqData := &adapters.RequestData{
+			Method:  "POST",
+			Uri:     fmt.Sprintf("%s?apptoken=%s&zoneid=%d", a.URI, pubnativeExt.AppAuthToken, pubnativeExt.ZoneID),
```

Comment:
Consider adding a query param encoder on the AppAuthToken string for safety. The zone should be fine since it's an int.

Reply:
Good point, I changed it to use Go's URL encoding to build the query string.